src/dmd/iasmdmd.d

```diff
@@ -1231,10 +1231,10 @@ opflag_t asm_determine_operand_flags(ref OPND popnd)
                     {
                         if (popnd.disp >= byte.min &&
                             popnd.disp <= byte.max)
-                            us = CONSTRUCT_FLAGS(OpndSize._8, _rel, _flbl,0);
-                        else if (popnd.disp >= short.min &&
-                            popnd.disp <= short.max && !global.params.is64bit)
-                            us = CONSTRUCT_FLAGS(OpndSize._16, _rel, _flbl,0);
+                            us = CONSTRUCT_FLAGS(OpndSize._32_8, _rel, _flbl,0);
+                        //else if (popnd.disp >= short.min &&
+                            //popnd.disp <= short.max && global.params.is16bit)
+                            //us = CONSTRUCT_FLAGS(OpndSize._16, _rel, _flbl,0);
```

Comment:
Just remove ?

Reply:
I wanted to leave a reminder if I ever ported it back to DMC. That piece is necessary for 16 bit assembler.